https://bugs.gentoo.org/388817
http://debbugs.gnu.org/8223
patch from upstream, revision 100514 in emacs-23 branch
* src/unexmacosx.c (copy_data_segment): Also copy __got section. (Bug#8223)

--- emacs-23/src/unexmacosx.c
+++ emacs-23/src/unexmacosx.c
@@ -822,6 +822,7 @@
 	}
       else if (strncmp (sectp->sectname, "__la_symbol_ptr", 16) == 0
 	       || strncmp (sectp->sectname, "__nl_symbol_ptr", 16) == 0
+	       || strncmp (sectp->sectname, "__got", 16) == 0
 	       || strncmp (sectp->sectname, "__la_sym_ptr2", 16) == 0
 	       || strncmp (sectp->sectname, "__dyld", 16) == 0
 	       || strncmp (sectp->sectname, "__const", 16) == 0
